src/mod/applications/mod_avmd/mod_avmd.c

```diff
@@ -326,7 +327,7 @@ static switch_status_t avmd_launch_threads(avmd_session_t *s, switch_core_sessio
         d->result = AVMD_DETECT_NONE;
         d->lagged = 1;
         d->lag = idx + 1;
-        switch_threadattr_create(&thd_attr, avmd_globals.pool);
+        switch_threadattr_create(&thd_attr, switch_core_session_get_pool(session));
         switch_threadattr_stacksize_set(thd_attr, SWITCH_THREAD_STACKSIZE);
         if (switch_thread_create(&d->thread, thd_attr, avmd_detector_func, d, switch_core_session_get_pool(session)) != SWITCH_STATUS_SUCCESS) {
```

Comment:
@dev-ryanc , do we use avmd_globals.pool elsewhere? Shouldn't we stick to avmd_globals in this scope?

Reply:
No, the detectors thread lifetime should be the same lifetime of the session. Allocating it to the global pool will remain into that pool until the lifetime of the avmd module.